packages/react-reconciler/src/ReactFiberScheduler.js

```diff
@@ -1490,7 +1494,29 @@ module.exports = function<T, P, I, TI, PI, C, CC, CX, PL>(
     }
   }
 
-  function performWorkOnRoot(root, expirationTime) {
+  function flushRoot(root: FiberRoot, expirationTime: ExpirationTime) {
+    invariant(
+      !isRendering,
+      'work.commit(): Cannot commit while already rendering. This likely ' +
+        'means you attempted to commit from inside a lifecycle method.',
+    );
+    // Perform work on root as if the given expiration time is the current time.
+    // This has the effect of synchronously flushing all work up to and
+    // including the given time.
+    performWorkOnRoot(root, expirationTime, expirationTime);
```

Comment:
@sebmarkbage This is the part we were discussing earlier. Need some way to tell the reconciler to treat all work up to `expirationTime` as synchronous. Here we do this by passing the same time as both the current time and the expiration time. But since the reconciler only uses the current time to check whether it's sync or async, a boolean would also work.